docs/user-guide/CAI.rst

```diff
@@ -0,0 +1,308 @@
+CAI allows training a harmless but non-evasive AI assistant that engages with harmful queries by explaining its objections to them.
+
+.. _Constitutional AI (CAI): https://arxiv.org/abs/2212.08073
+
+CAI
+###############
+The basic steps of CAI, as illustrated in Figure 1 of the paper (below):
+
+(Supervised Stage) Critique → Revision → Supervised Learning: The AI generates responses to harmfulness prompts using a helpful-only AI assistant, then critiques and revises its own responses according to a principle in the constitution, and then finetunes the original model on the revised responses.
+
+(RL Stage) AI Comparison Evaluations → Reward Model → Reinforcement Learning: The AI generates pairs of responses to harmfulness prompts using the finetuned model, then evaluates which response is better according to a principle in the constitution, and then trains a reward model based on this dataset of AI preferences and a human helpfulness preferences. The AI then trains with RL using the learned reward model.
+
+.. image:: ../assets/cai_diagram.png
+   :alt: basic steps of the CAI process
+
+Critiques, revisions and AI harmlessness feedback are steered by a small set of principles drawn from a ‘constitution’. The supervised stage significantly improves the initial model, and gives some control over the initial behavior at the start of the RL phase, addressing potential exploration problems. The RL stage significantly improves performance and reliability.
+
+Motivation
+###############
+
+Scaling supervision: using AI to help humans supervise other AIs more efficiently and effectively, especially for tasks where AI capabilities may exceed human ones.
+
+A harmless but non-evasive assistant: reducing the tension between helpfulness and harmlessness, and avoiding evasive responses that reduce transparency and helpfulness.
+
+Simplicity and transparency: encoding the training goals in a simple list of natural language instructions or principles, and using chain-of-thought reasoning to make AI decision making explicit and understandable.
+
+Reducing iteration time: obviating the need to collect new human feedback labels when altering the objective or testing different behaviors.
+
+Train a CAI model
+#####################
+
+This section is a step-by-step tutorial that walks you through how to run a full CAI pipeline with a ``Mistral-7B`` LLM model. It includes the following:
+
+1. Data download and preprocessing.
+
+2. Generate responses to harmfulness prompts using a helpful-only AI assistant. Ask the model to critique its response according to a principle in the constitution, and then revise the original response in light of the critique.
+
+3. Finetune ``Mistral-7B`` with SFT on the revised responses to create a ``Mistral-7B-SL-CAI`` model.
+
+4. 
```

Comment:
Delete 4.

Reply:
Steps 3 and 4 are two distinct stages. I have added the missing headline for steps a. b. and c. to step 4.